Src/BootCamp.Chapter/TextTable.cs

```diff
@@ -24,17 +26,62 @@ public static class TextTable
            | Hello |
            |       |
            +-------+
-
          */
 
         /// <summary>
         /// Build a table for given message with given padding.
         /// Padding means how many spaces will a message be wrapped with.
         /// Table itself is made of: "+-" symbols. 
         /// </summary>
+        
         public static string Build(string message, int padding)
         {
-            return "";
+            if (string.IsNullOrEmpty(message))
+            {
+                return "";
+            }
+            string[] textSplit = message.Split(new string[] { System.Environment.NewLine }, System.StringSplitOptions.None);
+            StringBuilder textTable = new StringBuilder();
+
+                int maxIndex = GetMaxIndex(textSplit);
+                int iOne;
+
+            textTable.Append("+").Append('-', textSplit[maxIndex].Length + padding * 2).AppendLine("+");
+            for (iOne = 0; iOne < padding; iOne++)
+            {
+                textTable.Append("|").Append(' ', textSplit[maxIndex].Length + padding * 2).AppendLine("|");
+            }
+            for (iOne = 0; iOne < textSplit.Length; iOne++)
+            {
+                textTable.Append("|").Append(' ', padding).Append(textSplit[iOne]);
+                textTable.Append(' ', textSplit[maxIndex].Length - textSplit[iOne].Length);
+                textTable.Append(' ', padding).AppendLine("|");
+            }
+            for (iOne = 0; iOne < padding; iOne++)
+            {
+                textTable.Append("|").Append(' ', textSplit[maxIndex].Length + padding * 2).AppendLine("|");
+            }
+            textTable.Append("+").Append('-', textSplit[maxIndex].Length + padding * 2).AppendLine("+");
+
+                return textTable.ToString();
```

Comment:
Split more into private functions so that individual steps are more highlighted. Right now I am burdened to read all until I can understand. Also, quite a lot of duplicate code, reuse it passing argument as a difference.

Reply:
Oh Gosh! I think I am going to to refactor step by step everything! Thanks to you I am learning a lot! Thanks Kai..